src/tests/java/picard/analysis/CollectWgsMetricsTest.java

```diff
@@ -0,0 +1,171 @@
+package picard.analysis;
+
+import htsjdk.samtools.SAMException;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMFileWriter;
+import htsjdk.samtools.SAMFileWriterFactory;
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMRecordSetBuilder;
+import htsjdk.samtools.metrics.MetricsFile;
+import htsjdk.variant.utils.SAMSequenceDictionaryExtractor;
+import org.testng.Assert;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import picard.cmdline.CommandLineProgramTest;
+import picard.sam.SortSam;
+
+import java.io.*;
+import java.util.Random;
+
+/**
+ * Tests for methods in CollectWgsMetrics
+ *
+ *
+ * @author Kylee Bergin
+ */
+
+public class CollectWgsMetricsTest extends CommandLineProgramTest {
+
+    private final static File DICT_DIR = new File("testdata/picard/sam/CollectGcBiasMetrics/");
+    private final static File TEST_DIR = new File("testdata/picard/sam/");
+    private final File dict = new File(DICT_DIR, "MSmallHeader.dict");
+    private File tempSamFile;
+    private File outfile;
+    private final static int LENGTH = 99;
+
+    private final static String sample = "TestSample1";
+    private final static String readGroupId = "TestReadGroup1";
+    private final static String platform = "ILLUMINA";
+    private final static String library = "TestLibrary1";
+    private final static int numReads = 40000;
+
+    public String getCommandLineProgramName() {
+        return CollectWgsMetrics.class.getSimpleName();
+    }
+
+    @DataProvider(name = "wgsDataProvider")
+    public Object[][] wgsDataProvider() {
+        final String referenceFile = "testdata/picard/quality/chrM.reference.fasta";
+
+        return new Object[][] {
+                {tempSamFile, outfile, referenceFile}
+        };
+    }
+
+    @Test(dataProvider = "wgsDataProvider")
+    public void testMetricsFromWGS(final File input, final File outfile, final String referenceFile) throws IOException {
+        outfile.deleteOnExit();
+        final String[] args = new String[] {
+                "INPUT="  + input.getAbsolutePath(),
+                "OUTPUT=" + outfile.getAbsolutePath(),
+                "REFERENCE_SEQUENCE=" + referenceFile
+        };
+        Assert.assertEquals(runPicardCommandLine(args), 0);
+
+        final MetricsFile<CollectWgsMetrics.WgsMetrics, Comparable<?>> output = new MetricsFile<CollectWgsMetrics.WgsMetrics, Comparable<?>>();
+        output.read(new FileReader(outfile));
+
```

Comment:
I'm not loving that these metrics required a tolerance for the assert equals, I think it's because there aren't enough reads in the sam file that was generated, so the mean coverage keeps coming out slightly different. I could make the sam file larger?